src/server/wesnothd/server.cpp

```diff
@@ -2131,6 +2134,49 @@ void server::requests_handler(const std::string& /*issuer_name*/,
 	metrics_.requests(*out);
 }
 
+void server::roll_handler(const std::string& issuer_name,
+		const std::string& /*query*/,
+		std::string& parameters,
+		std::ostringstream* out)
+{
+	assert(out != nullptr);
+	if(parameters.empty()) {
+		return;
+	}
+	
+	int N;
+	try {
+		N = std::stoi(parameters);
+	} catch(const std::invalid_argument&) {
+		*out << "The number of die sides must be a number!";
+		return;
+	} catch(const std::out_of_range&) {
+		*out << "The number of sides is too big for the die!";
+		return;
+	}
+
+	if(N < 1) {
+		*out << "The die cannot have less than 1 side!";
+		return;
+	}
+	std::uniform_int_distribution<int> dice_distro(1, N);
+	std::string value = std::to_string(dice_distro(die_));
+
+	*out << "You rolled a die [1 - " + parameters + "] and got a " + value + ".";
+
+	auto player_ptr = player_connections_.get<name_t>().find(issuer_name);
+	if(player_ptr == player_connections_.get<name_t>().end()) {
+		return;
+	}
+
+	auto g_ptr = player_ptr->get_game();
+	if(g_ptr && g_ptr->is_member(player_ptr->socket())) {
```

Comment:
The `is_member` call seems useless. If `g_ptr` is valid the player is part of the game.

Reply:
It's being also checked at
 https://github.com/wesnoth/wesnoth/blob/680559037a4145be376833b0e055cf0e6ac2369a/src/server/wesnothd/server.cpp#L1126-1130 , I'm not quite sure about this.

Reply:
Yes, it is useless there as well. `is_member` checks for players or observers. There is nothing more in a game.

Reply:
Fair enough, put the fix in a new commit just in case.